abi/codecForCompositeTypes_list.go

```diff
@@ -0,0 +1,83 @@
+package abi
+
+import (
+	"bytes"
+	"errors"
+	"io"
+)
+
+func (c *codec) encodeNestedList(writer io.Writer, value InputListValue) error {
```

Comment:
Indeed. Change in progress.

Reply:
Refactored, using separate structs now. Using regular composition, though, instead of go-like composition. Some codecs rely on the general codec to perform their tasks.

I've postponed the renaming of the files, so that this fix-after-review doesn't bring a big delta: https://github.com/multiversx/mx-sdk-abi-go/issues/3.